contracts/src/BattleShip.sol

```diff
@@ -5,13 +5,22 @@ import "./interfaces/IGame.sol";
 
 contract BattleShip is IGame {
     uint256 public number;
+    bytes32 public digest;
+
+    function initializeGame(uint256 gameId) external {
+        emit Initialize(gameId);
+    }
+
+    function gameInitialized(uint256 gameId, bytes32 _digest) external {
```

Comment:
why do we need 2 methods for game init? In case we let the client to request a game start, it _might_ improve user experience, but I believe we can do without it in the hackathon version.